docs/source-pytorch/common/trainer.rst

```diff
@@ -1629,6 +1629,18 @@ The number of epochs run.
     if trainer.current_epoch >= 10:
         ...
 
+datamodule
+**********
+
+The current datamodule, which is used by the trainer. It shouldn't be replaced by another datamodule object.
```

Comment:
We should drop the remark about not replacing it, and instead make the attribute read-only. This way, it won't be possible for a user to do that (it will show an error).

```suggestion
The current datamodule, which is used by the trainer.
```

Reply:
@awaelchli so this is more a code issue now?

Reply:
Yes I think it is a good idea to do both. It is good to document the datamodule here, but not many people will go read it first before using it. So it is better to handle the misuse directly by an error in the code. Let's follow up on this after we add the docs here?